core/record/_record/recordNodesUpdater.js

```diff
@@ -69,6 +69,7 @@ const _addOrUpdateAttribute =
         valueSearch: value,
         record,
         parentNode: entity,
+        strict: true,
```

Comment:
I should add a comment on this. Basically when strict=true it compares the internal uuid in the values (e.g. the itemUuid of a code attribute value), instead when it's false it can determine the itemUuid corresponding to a value in another format, like just category item code (this happens when expressions are evaluated).